.travis.yml

```diff
@@ -57,6 +57,14 @@ script:
   - if [ $REPO_PROJ == "qemu_v8" ]; then cd $HOME/$REPO_PROJ/qemu && git submodule update --init dtc; fi
   # Here we are using Travis environment variables to select the correct branch
   # based on either a pull request or a normal push.
-  - cd $HOME/$REPO_PROJ/build && git remote add github https://github.com/$TRAVIS_REPO_SLUG && git remote update
-  - cd $HOME/$REPO_PROJ/build && git checkout github/$TRAVIS_BRANCH
+  - |
+        if [ "$TRAVIS_EVENT_TYPE" == "pull_request" ]; then
```

Comment:
Actually it can trigger two builds when pushing a single update. I.e., both a local push on your own personal tree, but also one for the PR, i.e., the pull request it will still be triggered as a pull request.

> it's not clear from from description

Totally agree, Travis is a lot of guess work!